components/nexus-core/src/main/java/org/sonatype/nexus/proxy/repository/RepositoryStatusCheckerThread.java

```diff
@@ -53,6 +60,18 @@ public void run() {
     boolean interrupted = false;
 
     while (isRunning() && getRepository().getProxyMode() != null) {
+
+      // wait for Nexus to fully start
+      if (!systemStatus.isNexusStarted()) {
+        try {
+          Thread.sleep(10 * 1000);
```

Comment:
Each proxy repo has one of these threads, so for large configurations a very small interval here could be counter-productive. Also the main interval of these checker threads once started is typically set to 1 hour, so delaying them post-startup by at most 10s is not a big deal in the overall context of what they're doing. Were you thinking of 1s, 3s, 5s?